test/encapsulate_op/encapsulate_op_test.cc

```diff
@@ -0,0 +1,192 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "tensorflow/core/graph/node_builder.h"
+
+#include "../test_utilities.h"
+#include "ngraph_backend_manager.h"
+#include "ngraph_encapsulate_impl.h"
+#include "ngraph_encapsulate_op.h"
+#include "ngraph_utils.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+namespace testing {
+#define ASSERT_OK(x) ASSERT_EQ((x), ::tensorflow::Status::OK());
+#define ASSERT_NOT_OK(x) ASSERT_NE((x), ::tensorflow::Status::OK());
+
+// Test: Create tensorflow input tensors and Compute Signature
+TEST(EncapsulateOp, ComputeSignature) {
```

Comment:
This is a good test but in future we need to expand this to testing with different corner
cases. In that case we can load pbtxt graph snippets with complex signatures and compare them with expected results stored in files.